### DISCUSSION TITLE
Kafka Scaler: Split `tls` section into "tls related things" and certificate authentication

### DISCUSSION THREAD

**Discussion by JorTurFer** [kedacore/keda#2617] [General] [answered]
Right now, the certificate authentication parameters are inside `tls` section. Usually `tls` is related with the transport layer and not with the authentication method. This could be confusing because if you use user/pass authentication you still enable `tls` section which also contains the data for certificate authentication.

My proposal is:

- Enable secure connections using a new parameter named `useTls`.
- Add a new parameter `unsafeSsl` for trusting on server certificate.
- Create a new section named `certificate` for all the data related with the certificate.
- Deprecate current `tls` section and keep as it is now for some version and then remove it.

**Comment by tomkerkhove** [comment, marked as answer]
`certificate` would represent, what? certificate authentication or?

What if `unsafeSsl` is `disableTlsValidation` or something more explicit?

**Reply by JorTurFer** [reply]
[now `tls` contains these parameters](https://keda.sh/docs/2.6/scalers/apache-kafka/#authentication-parameters):

- tls - To enable SSL auth for Kafka, set this to enable. If not set, TLS for Kafka is not used. (Values: true, false, Default: false, Optional)
- ca - Certificate authority file for TLS client authentication. (Optional)
- cert - Certificate for client authentication. (Optional)
- key - Key for client authentication. (Optional)

They are under `Authentication Parameters` section.
My proposal is moving the 3 latest into a new `certificate` section inside `Authentication Parameters` and keep the `tls` parameter outside that section

`certificate`:
- ca - Certificate authority file for TLS client authentication. (Optional)
- cert - Certificate for client authentication. (Optional)
- key - Key for client authentication. (Optional)

Related with `unsafeSsl`. This parameter is already in use on several scalers, that's why I'd use the same to be consistent

**Reply by JorTurFer** [reply]
Maybe I'm overthinking about this 😢

**Reply by zroubalik** [reply]
I am against this. It is a breaking change.

And the current approach is ok. Do you want to use tls? Enable tls.
Do you want to supply your cert/ca/key? Just specify them.

And as I mentioned in the original issue, I think this is sort of aligned with Kafka world. And I think we should be consistent, so users are familiar with the nomenclature for the technology.

**Reply by JorTurFer** [reply]
so yes, I was overthinking 😄

**Reply by zroubalik** [reply]
No worries, it is a good practice to try to find improvements on existing stuff!
